### PR TITLE
Collapsing TranslationRequest -> ResponseOptions

### PR DESCRIPTION
**Pull request by jerinphilip**
Towards #77.

`QualityScore.h` was only used by `TranslationRequest` before (after `TranslationResult` merged into `Response`). This can now be deleted (https://github.com/browsermt/bergamot-translator/pull/109#discussion_r623957470).

For a first-step `TranslationRequest` is typedef-ed as `ResponseOptions` to see if it's a clean collapse.

TODO:
- [x] Copy over salvageable documentation

Edit: I figure the following can be a separate PR, idea is configurability of responseOptions out so extension can iterate. Choose one of the following.

  - [x] **A**: Add a JSON/YAML constructor using marian-internals. This will need to be replaced with a faster parsing mechanism or direct access to the struct containing `ResponseOptions/TranslationRequest`. (#141)
  - [x] **B**: Provide struct TranslationRequest access directly to WASM (#140)
  
Unsure if both **A** and **B** can be retained.

### CODE REVIEW

File: wasm/bindings/TranslationRequestBindings.cpp [hunk 1]
```diff
@@ -5,7 +5,9 @@
 
 #include <emscripten/bind.h>
 
-#include "TranslationRequest.h"
+#include "response_options.h"
+
+typedef marian::bergamot::ResponseOptions TranslationRequest;
```
Comment (XapaJIaMnu) [inline wasm/bindings/TranslationRequestBindings.cpp:10]: Bad. Refactor cleanly.

**Review by XapaJIaMnu — CHANGES_REQUESTED** [review]
No typedef hacks please. If we can't change it because we're breaking the plugin, negotiate the change with the plugin. Else don't change it.

File: wasm/bindings/TranslationRequestBindings.cpp (hunk 1 above)
Comment (jerinphilip) [reply wasm/bindings/TranslationRequestBindings.cpp:10]: I have to draw the line somewhere - if I have to make it clean I have to clean it in extension as well by the rationale. This is the cleanest it in my perspective, so the binding semantics below stay consistent (TranslationRequest -> "TranslationRequest") in JS. The plugin can do their part later? 

My objective here is to get the two different entities "TranslationRequest" | "ResponseOptions" into one of Service so incoming concurrency changes don't wonder why there exist two.

File: src/translator/service.h [hunk 2]
```diff
@@ -137,13 +136,13 @@ class Service {
...
-  /// indicating whether or not to include some member in the Response, also
-  /// specify any additional configurable parameters.
+  /// @param [in] translationRequest: ResponseOptions indicating whether or not
+  /// to include some member in the Response, also specify any additional
+  /// configurable parameters.
 
   std::vector<Response>
   translateMultiple(std::vector<std::string> &&source,
-                    TranslationRequest translationRequest);
+                    ResponseOptions responseOptions);
```
[the remark this answers is not in the document]
Comment (jerinphilip) [reply src/translator/service.h:145]: bools and enums only as of now, about 5. WASM previously was pass by value so I'm keeping the signature.

File: wasm/bindings/TranslationRequestBindings.cpp (hunk 1 above)
Comment (abhi-agg) [reply wasm/bindings/TranslationRequestBindings.cpp:10]: Created a separate issue https://github.com/browsermt/bergamot-translator/issues/149 for this.

**Comment by abhi-agg** [comment]
> No typedef hacks please. If we can't change it because we're breaking the plugin, negotiate the change with the plugin. Else don't change it.

Created a separate issue https://github.com/browsermt/bergamot-translator/issues/149 to address clean refactoring

**Review by abhi-agg — APPROVED** [review]

**Comment by XapaJIaMnu** [comment]
@abhi-agg before we approve this, can you confirm that we have agreed on the same name at #149 ?

**Comment by XapaJIaMnu** [comment]
Ping @abhi-agg

**Comment by abhi-agg** [comment]
> Ping @abhi-agg

@XapaJIaMnu I suggested few options there and we need to decide which one we opt for. I pinged everyone again. It is a decision that all of us need to take together 👍

**Comment by XapaJIaMnu** [comment]
Jerin has proposed name here, I think that name is fine, if you agree I can merge.

**Comment by abhi-agg** [comment]
> Jerin has proposed name here, I think that name is fine, if you agree I can merge.

@XapaJIaMnu Sorry but I am a bit lost here. I saw only 1 comment from Kenneth on the issue. Did I miss something?

**Comment by XapaJIaMnu** [comment]
`ResponseOptions` got renamed to `TranslationRequest` other bits of the code should follow so we can remove the `typedef`

**Review by XapaJIaMnu — APPROVED** [review]
Ok, I guess let's start renaming. Let's get rid of the `typedef` ASAP. @abhi-agg please take the necessary steps to rename things in the other repos.

**Comment by abhi-agg** [comment]
> `ResponseOptions` got renamed to `TranslationRequest` other bits of the code should follow so we can remove the `typedef`

@XapaJIaMnu Do you mean, `ResponseOptions` is renamed back to `TranslationRequest` in this PR? I don't see it here.

> Ok, I guess let's start renaming. Let's get rid of the `typedef` ASAP. @abhi-agg please take the necessary steps to rename things in the other repos.

I don't see a clear decision. Which [option](https://github.com/browsermt/bergamot-translator/issues/149#issue-890174034) did we decide on finally?

**Comment by XapaJIaMnu** [comment]
TranslationRequest got renamed to response options

**Pull request merged**